tools/scale_image/scale_image.py

```diff
@@ -21,24 +21,25 @@ def scale_image(input_file, output_file, scale, order=1):
             interp = 'bicubic'
 
         if ',' in scale:
-            scale = scale.replace('[','').replace(']','').split(',').reverse()
+            scale = scale[1:-1].split(',')
+            scale.reverse()
             scale = [int(i) for i in scale]
         elif '.' in scale:
             scale = float(scale)
         else:
             scale = int(scale)
 
         res = scipy.misc.imresize(img_in, scale, interp=interp)
-        skimage.io.imsave("tmp.png", res)
-        os.rename("tmp.png", output_file)
+        png_file = output_file.replace('.dat', '.png')# otherwise we get problems with the .dat extension of galaxy
```

Comment:
I'm not sure this is correct. Galaxy does not store anything in filenames (because if can deal with object-stores as well), so every Galaxy file object has no ending or `.dat`. If you change the output_file path (I assume that is the Galaxy path e.g. /database/files/000/0002021.dat) to `png` Galaxy will not find the output.
The previous code does look good to me or you can always do this on the Galaxy part and not inside this script.

```
         <![CDATA[
         python '$__tool_directory__/scale_image.py' '$input' ./out.png
         #if $use_scale_option.use_scale == 'true'
           $use_scale_option.scale
         #else if $use_scale_option.use_scale == 'false'
           '[$use_scale_option.scale_x, $use_scale_option.scale_y]'
         #end if
         $order

        &&
        mv ./out.png '$output'
        ]]>
```